File: src/longest-consecutive-sequence/longest-consecutive-sequence.ts
function longestConsecutive(nums: number[]): number {
    if (nums.length === 0) return 0;

    const numSet = new Set(nums);
    let maxGlobal = 0;

    for (const num of numSet) {
        if (!numSet.has(num - 1)) {
            let maxCurr = 1;
            let nextNum = num + 1;

            while (numSet.has(nextNum)) {
                maxCurr++;
                nextNum++;
            }

            maxGlobal = Math.max(maxGlobal, maxCurr);
        }
    }

    return maxGlobal;
};

File: src/longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0
        numset = set(nums)
        max_global = 0
        for num in numset:
            if num - 1 not in numset:
                max_curr = 1
                next_num = num + 1
                while (next_num in numset):
                    max_curr+=1
                    next_num+=1
                max_global = max(max_global, max_curr)
        return max_global